boards/st/nucleo_u031r8/nucleo_u031r8.dts

```diff
@@ -81,3 +81,12 @@
 	ahb-prescaler = <1>;
 	apb1-prescaler = <1>;
 };
+
+&adc1 {
+	pinctrl-0 = <&adc1_in0_pc0 &adc1_in1_pc1>;
+	pinctrl-names = "default";
+	st,adc-clock-source = <ASYNC>;
```

Comment:
You set the clock source as asynchronous, but you didn't explicitly select which clock you wanted. It works but it is not good practice.
Either use SYNC, or define a secondary source clock.
To repeat on all three boards